src/main/java/fi/fmi/avi/archiver/message/populator/ValidityPeriodPopulator.java

```diff
@@ -0,0 +1,36 @@
+package fi.fmi.avi.archiver.message.populator;
+
+import fi.fmi.avi.archiver.file.InputAviationMessage;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+
+import javax.annotation.Nullable;
+import java.time.Duration;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ValidityPeriodPopulator implements MessagePopulator {
```

Comment:
How about something a bit more descriptive name, e.g. `FixedDurationValidityPeriodPopulator` or something?

Reply:
Perhaps just FixedValidityPeriodPopulator? The duration part seems implied in a period.

Reply:
Thinking about this more I guess `FixedDurationValidityPeriodPopulator` is good as it's more explicit. We could theoretically have other populators where the period is fixed but not based on a duration.

Reply:
That's what I was also thinking, that the period is based on a fixed duration.